lib/solidus_affirm_v2/callback_hook/base.rb

```diff
@@ -17,7 +17,7 @@ def authorize!(payment)
           }
         )
 
-        remove_tax!(payment.order) if authorized_affirm.provider == "katapult"
+        remove_tax!(payment.order) if authorized_affirm.provider_id == :katapult
```

Comment:
provider_id is either 1, or 2. This is fixed in the branch already. Thanks a lot for opening a PR though! Really appreciated!